src/components/Swap/SwapActionButton/index.tsx

```diff
@@ -36,7 +36,7 @@ export default function SwapActionButton() {
 
   if (!account || !isActive) {
     return <ConnectWalletButton />
-  } else if (error === ChainError.MISMATCHED_CHAINS) {
+  } else if (error === ChainError.MISMATCHED_CHAINS || error === ChainError.UNSUPPORTED_CHAIN) {
     const tokenChainId = inputCurrency?.chainId ?? outputCurrency?.chainId ?? SupportedChainId.MAINNET
```

Comment:
Unless it's happening elsewhere, I think you'll probably want to confirm that the input/output currency chainId is supported instead of trusting it to be supported if it is defined.